core/oputil/oputil_ssh_key.js

```diff
@@ -0,0 +1,158 @@
+            message: 'Generate New SSH Keys?',
+            type: 'confirm',
+            default: false,
+        },
+        {
+            name: 'password',
+            message: 'SSL Password:',
+            default: "",
+            when: answers => answers.createNew,
+        },
+    ],
+};
+
+function execute(ui, command) {
+    ui.log.write("Ping!");
+    ui.log.write(command);
+    exec(
+        command,
+        function (error, stdout, stderr) {
+            ui.log.write(error);
+
+            if (error) {
+                const reason = error ? error.message : 'OpenSSL Error';
+                logDebug(
+                    {
+                        reason: reason,
+                        cmd: util.cmd,
+                        args: args
+                    },
+                    `openssl command failed`
+                );
+            }
+            else {
+                ui.log.write("SSH Keys Generated")
+            }
+        }
+    );
+}
+
+function createNew(cb) {
+    const ui = new inq.ui.BottomBar();
+
+    let sslPassword;
+
+    async.waterfall(
+        [
+            function init(callback) {
+                return initConfigAndDatabases(callback);
+            },
+            function create(configuration, callback) {
+                getAnswers(QUESTIONS.Create, answers => {
+                    if (!answers.createNew) {
+                        return callback('exit');
+                    }
+
+                    // Get Answer Value
+                    sslPassword = answers.password;
+                    if (!sslPassword || sslPassword.replaceAll(" ", "") == "") {
+                        ui.log.write('Password must be set.');
+
+                        return callback('exit');
+                    }
+                    if (sslPassword.length < MINIMUM_PASSWORD_LENGTH) {
+                        ui.log.write(`Password must be at least ${MINIMUM_PASSWORD_LENGTH} characters.`);
+
+                        return callback('exit');
+                    }
+
+                    // Check if Keyfiles Exist
+                    const sshKeyPath = "config/security/";
+                    const sshKeyFilename = "ssh_private_key.pem";
+                    const targetKeyFile = sshKeyPath + sshKeyFilename;
+
+                    // Check if Keyfile Exists
+                    if (fs.existsSync(targetKeyFile)) {
+                        ui.log.write(`${targetKeyFile} already exists.`)
+
+                        return callback('exit');
+                    }
+
+                    ui.log.write(`Creating SSH Key: ${targetKeyFile}`);
+
+                    // Create Dir
+                    if (!fs.existsSync(sshKeyPath)) {
+                        ui.log.write(`Creating Directory: ${sshKeyPath}`);
+                        exec(`mkdir -p ${sshKeyPath}`);
+                    }
+
+                    // Check if OpenSSL binary is installed
+                    const binaryPath = "/usr/bin/openssl";
```

Comment:
No need to check if a binary exists before trying it, just try it. You'll get a not found error or the like. 

I'm wondering if we should just use "openssl" here, since `/usr/bin/` isn't always it's home

Reply:
Done

I'm used to always doing checks such as these in my day job, because it is my experience that smart people will spend hours burning away their mental CPU power on trivial things that could have otherwise been made obvious to them.

As an upcoming recommendation, I would like to implement a pre-flight check routine in the BBS Startup. It would:
- Validate all paths, filenames, and binary references in config.hjson
- Check if ports are already in use for content and login servers
- Produce actionable feedback so that a newbie / returning SysOp has a good experience
- Probably some other stuff, but this seems like a good start

Reply:
Certainly check the error code, this just eliminates a pre-condition that can succeed while the real create/open may still fail.

That all sounds great! For pre-checks, if you want to tackle it, a JSON schema for the config could do most of the work, see https://github.com/NuSkooler/enigma-bbs/issues/281